src/worker/runner/test-runner/index.js

```diff
@@ -21,40 +22,107 @@ module.exports = class TestRunner extends Runner {
         super();
 
         this._test = test.clone();
-        this._test.testplaneCtx = _.cloneDeep(test.testplaneCtx);
+        this._test.testplaneCtx = _.cloneDeep(test.testplaneCtx) || {};
 
         this._file = file;
         this._config = config;
         this._browserAgent = browserAgent;
     }
 
-    async run({ sessionId, sessionCaps, sessionOpts, state, ExecutionThreadCls = ExecutionThread }) {
-        const test = this._test;
-        const testplaneCtx = test.testplaneCtx || {};
+    async run({ sessionId, sessionCaps, sessionOpts, state }) {
+        await this.prepareToRun({ sessionId, sessionCaps, sessionOpts, state });
+
+        const error = await this.runRunnables(ExecutionThread);
 
-        let browser;
+        return this.finishRun(error);
+    }
+
+    // TODO: make it protected
+    async prepareToRun({ sessionId, sessionCaps, sessionOpts, state }) {
```

Comment:
I want to use these methods in `BrowserTestRunner` and stub them. So I want make these methods protected but it is on javascript. Therefore, I left todo here for now